### PR TITLE
feat(pie-button): DSW-867 added slots for leading and trailing icons

### PR DESCRIPTION
**Pull request by ashleynolan**
### `pie-button`

- [Added] - Slots for leading and trailing icons

### `pie-storybook`

- [Added] - expanded controls info and pie-button updated to include icon slots

### `wc-vanilla` example app
- [Added] - Leading & Trailing Button Icons and Icon Button examples

## Author Checklist
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [x] If it is a component change, I have reviewed the Storybook preview
- [x] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-storybook/stories/pie-button.stories.ts [hunk 1]
```diff
@@ -16,35 +19,69 @@ const defaultArgs: ButtonProps = {
...
         size: {
+            description: 'Set the size of the button',
             control: 'select',
             options: sizes,
+            defaultValue: {
+                summary: 'medium',
+            },
         },
         type: {
+            description: 'Set the type the button,',
```
Comment (LTurns) [inline apps/pie-storybook/stories/pie-button.stories.ts:38]:
```suggestion
            description: 'Set the type of the button.',
```

File: apps/pie-storybook/.storybook/preview.js [hunk 2]
```diff
@@ -55,6 +55,9 @@ export default {
                     value: '#262626',
                 },
             ]
+        },
+        controls: {
+        expanded: true,
```
Comment (LTurns) [inline apps/pie-storybook/.storybook/preview.js:60]: super minor - line indent needed here I think

File: packages/components/pie-button/src/button.scss [hunk 3]
```diff
@@ -278,3 +293,9 @@
         }
     }
 }
+
+// Used to size an SVG if one is passed in (so not using the component icons
```
Comment (fernandofranca) [inline packages/components/pie-button/src/button.scss:297]: Tiny nit: missing closing parenthesis

File: packages/components/pie-button/src/index.ts [hunk 4]
```diff
@@ -16,6 +16,11 @@ export {
 
 const componentSelector = 'pie-button';
 
+/**
+ * @slot icon-leading - Leading icon
```
Comment (fernandofranca) [inline packages/components/pie-button/src/index.ts:20]: Were these comments added for a specific reason? I'm trying to figure out other scenarios where this applies.

Comment (jamieomaguire) [reply packages/components/pie-button/src/index.ts:20]: It allows the `custom-elements.json` file to correctly list any slots used by the component. I do not think the CEM analyser picks them up by default.

File: apps/pie-storybook/stories/pie-button.stories.ts (hunk 1 above)
Comment (ashleynolan) [reply apps/pie-storybook/stories/pie-button.stories.ts]: Fixed

File: apps/pie-storybook/.storybook/preview.js (hunk 2 above)
Comment (ashleynolan) [reply apps/pie-storybook/.storybook/preview.js]: Fixed

File: packages/components/pie-button/src/button.scss (hunk 3 above)
Comment (ashleynolan) [reply packages/components/pie-button/src/button.scss]: Fixed

File: apps/pie-storybook/stories/pie-button.stories.ts [hunk 5]
```diff
@@ -16,35 +19,69 @@ const defaultArgs: ButtonProps = {
...
 const buttonStoryMeta: ButtonStoryMeta = {
     title: 'Button',
     component: 'pie-button',
     argTypes: {
         size: {
+            description: 'Set the size of the button.',
             control: 'select',
             options: sizes,
+            defaultValue: {
+                summary: 'medium',
```
Comment (raoufswe) [inline apps/pie-storybook/stories/pie-button.stories.ts:34]: just curious, what does summary do here?

**Comment by raoufswe** [comment]
i might be missing something, but the icons are're rendering in the deployed storybook. 

Screenshot 2023-07-25 at 09 49 51

**Comment by jamieomaguire** [comment]
> i might be missing something, but the icons aren't rendering in the deployed storybook.
> 
> Screenshot 2023-07-25 at 09 49 51

@raoufswe I've mentioned to the team that currently Storybook is treeshaking the icons in production deploys. So unfortunately you won't see them on the deploy and will need to run locally if you want to see the changes.

Myself and Xander have a fix in progress so this won't be an issue for very long :)

File: apps/pie-storybook/.storybook/preview.js [hunk 6]
```diff
@@ -55,6 +55,9 @@ export default {
                     value: '#262626',
                 },
             ]
+        },
+        controls: {
+            expanded: true,
```
Comment (jamieomaguire) [inline apps/pie-storybook/.storybook/preview.js:60]: What does this do, out of interest?

**Review by jamieomaguire — APPROVED** [review]

File: apps/pie-storybook/.storybook/preview.js (hunk 6 above)
Comment (xander-marjoram) [reply apps/pie-storybook/.storybook/preview.js:60]: That shows the description and default value columns in the Controls tab

**Review by raoufswe — APPROVED** [review]

File: apps/examples/wc-vanilla/index.html [hunk 7]
```diff
@@ -34,5 +34,7 @@
         <script type="module" src="/main.js"></script>
         <script type="module" src="./node_modules/@justeattakeaway/pie-button"></script>
         <script type="module" src="./node_modules/@justeattakeaway/pie-icon-button"></script>
+        <script type="module" src="./node_modules/@justeattakeaway/pie-icons-webc/icons/IconClose"></script>
```
Comment (xander-marjoram) [inline apps/examples/wc-vanilla/index.html:37]: @jamieomaguire What do we think about trying to keep the path the same in our PR so we don't have to update all of these?

File: apps/pie-storybook/stories/pie-button.stories.ts (hunk 5 above)
Comment (ashleynolan) [reply apps/pie-storybook/stories/pie-button.stories.ts:34]: It's basically the only way to add the `defaultValue` in our storybook. Without this syntax (so just assigning a string for instance) the default value doesn't display.

Even stranger than that – `summary` is the only available property you can set in this object 😆

File: apps/pie-storybook/.storybook/preview.js (hunk 6 above)
Comment (ashleynolan) [reply apps/pie-storybook/.storybook/preview.js:60]: What Xander said – I've added the detail for the button (as that was the component I was updating). Would be good to do the same for the other components too as we update them (as doesn't take v long)

File: apps/examples/wc-vanilla/index.html (hunk 7 above)
Comment (ashleynolan) [reply apps/examples/wc-vanilla/index.html:37]: If you can, that would be great 👍

**Pull request merged**